### ISSUE TITLE
Bug: nixStatic has dependencies in /nix/store

### ISSUE THREAD

**Issue by michaelCTS** [NixOS/nixpkgs#262007]
### Describe the bug

When trying to run `nix-channel` using the static binary in a virgin, non-nix environment, it fails trying to find `nix-env` in `/nix/store`. `store = /home/user/my-store` in `nix.conf` is ignored.

### Steps To Reproduce
1. Start a virgin, non-nix environment e.g alpine or debian in a VM or docker
2. Download the static nix binary (or transfer it from your nix environment) (see [discourse for attempts at making that easier](https://discourse.nixos.org/t/where-can-i-get-a-statically-built-nix/34253) and https://github.com/NixOS/nix/issues/9176; a feature request for easier retrieval)
3. Create a folder and add it to `PATH`
4. Move the nix binary into it and create symlinks to it called `nix-channel` and `nix-env`
5. run `nix-channel --add https://nixos.org/channels/nixpkgs-unstable nixpkgs`
6. run `nix-channel --update`

### Expected behavior

The `nixpkgs` channel should be downloaded.

### Screenshots

```bash
$ nix-channel --update
unpacking channels...
error: executing '/nix/store/qibdl14c7jcxna48v5h5g51qm1iw2iw9-nix-static-x86_64-unknown-linux-musl-2.17.0/bin/nix-env': No such file or directory
error: program '/nix/store/qibdl14c7jcxna48v5h5g51qm1iw2iw9-nix-static-x86_64-unknown-linux-musl-2.17.0/bin/nix-env' failed with exit code 1
```

### Additional context

Basically was trying to do what @matthewbauer did https://matthewbauer.us/blog/static-nix.html , but with the official `nixStatic`. Related PR #56281 which was superseded by #70024

### Notify maintainers

@eelco @lovesegfault @artturin

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

--> this doesn't work as channels don't work

**Comment by Artturin** [comment]
This is a nix issue, not a nixpkgs one

https://github.com/NixOS/nix/blob/e58566a057692bbfbd30a6639248adfcf81df108/src/nix-channel/nix-channel.cc#L146

Try `export NIX_BIN_DIR=$(dirname $(type -p nix))`

weirdly the variable is not documented anywhere nor is it mentioned in any issue

https://github.com/search?q=%2F%5CbNIX_BIN_DIR%2F+NOT+is%3Afork&type=code

Maybe the code could be changed to re-execute itself with a different argv0?

**Comment by Artturin** [comment]
Please create a issue in the nix repo and I will report my comment there.

**Comment by Artturin** [comment]
Created https://github.com/NixOS/nix/issues/9193
